### DISCUSSION TITLE
How to call truffle Migrate from javascript instead of command line

### DISCUSSION THREAD

**Discussion by csirak** [ConsenSys-archive/truffle#5725] [General Q&A]
I want to be able to deploy contracts once I get a request from a user but truffle seems only to allow deployment from the command line, not a JS script. Thanks for the help!

**Comment by apdan510** [comment]
@csirak1528 could you please share the js script with us? Can we see what function executes the deployment? 

Thank you,
Alex | Truffle Support | Consensys

**Reply by csirak** [reply]
I figured it out. I just looked at the truffle core library and figured out how specific commands like 'deploy' or 'migrate' and also passing in specific args. I can send the script here if you'd like
